### PR TITLE
Bugfix in recarray serialization

### PR DESCRIPTION
**Pull request by dimitri-yatsenko**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ services:
   db:
     image: datajoint/mysql
     environment:
-      - MYSQL_ROOT_PASSWORD=simple
+      - MYSQL_ROOT_PASSWORD=""
```
Comment (guzman-raphael) [inline docker-compose.yml]:
```suggestion
      - MYSQL_ROOT_PASSWORD=
```

File: tests/__init__.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 CONN_INFO_ROOT = dict(
     host=environ.get('DJ_HOST', 'localhost'),
     user=environ.get('DJ_USER', 'root'),
-    password=environ.get('DJ_PASS', ''))
+    password=environ.get('DJ_PASS', 'simple'))
```
Comment (guzman-raphael) [inline tests/__init__.py]:
```suggestion
    password=environ.get('DJ_PASS', ''))
```

File: tests/__init__.py [hunk 3]
```diff
@@ -37,8 +37,10 @@
 PREFIX = environ.get('DJ_TEST_DB_PREFIX', 'djtest')
 
 conn_root = dj.conn(**CONN_INFO_ROOT)
+conn_root.query("DROP USER IF EXISTS 'datajoint'@'%%';") 
 conn_root.query("CREATE USER 'datajoint'@'%%' IDENTIFIED BY 'datajoint';")
 conn_root.query("GRANT ALL PRIVILEGES ON `djtest%%`.* TO 'datajoint'@'%%';")
+conn_root.query("DROP USER IF EXISTS 'djview'@'%%';") 
 conn_root.query("CREATE USER 'djview'@'%%' IDENTIFIED BY 'djview';")
 conn_root.query("grant select on `djtest%%`.* to 'djview'@'%%';")
```
Comment (guzman-raphael) [inline tests/__init__.py]: To update for compatibility between MySQL 5.6/5.7/8, can replace Lines 40-45 with: 

```
# create user if necessary on mysql8
if StrictVersion(conn_root.query(
        "select @@version;").fetchall()[0][0]) >= StrictVersion('8.0.0'):
    conn_root.query("CREATE USER IF NOT EXISTS 'datajoint'@'%%';")
    conn_root.query("CREATE USER IF NOT EXISTS 'djview'@'%%';")

# grant permissions and create users for mysql5.6/5.7
conn_root.query("GRANT ALL PRIVILEGES ON `djtest%%`.* \
        TO 'datajoint'@'%%' IDENTIFIED BY 'datajoint';")
conn_root.query("GRANT SELECT ON `djtest%%`.* \
        TO 'djview'@'%%' IDENTIFIED BY 'djview';")
```

Comment (dimitri-yatsenko) [reply tests/__init__.py]: ok, included. thanks.

**Pull request merged**